core/trino-main/src/main/java/io/trino/operator/scalar/ArrayConstructor.java

```diff
@@ -101,7 +105,16 @@ protected SpecializedSqlScalarFunction specialize(BoundSignature boundSignature)
         }
         ImmutableList<Class<?>> stackTypes = builder.build();
         Class<?> clazz = generateArrayConstructor(stackTypes, type);
-        MethodHandle methodHandle = methodHandle(clazz, "arrayConstructor", stackTypes.toArray(new Class<?>[stackTypes.size()]));
+        MethodHandle methodHandle;
+        if (stackTypes.size() >= 255) {
+            // varags
+            methodHandle = methodHandle(Stream.of(clazz.getMethods())
```

Comment:
There is, i guess, no benefit from this code being generated.
Perhaps we could just hook up a handle to ordinary method ("intrinsic").

Still might be value from perspective of POC-ing vararg calls.